packages/completer/src/manager.ts

```diff
@@ -119,17 +123,41 @@ export class CompletionProviderManager implements ICompletionProviderManager {
     const { widget, editor } = newCompleterContext;
     const id = widget.id;
     const handler = this._panelHandlers.get(id);
+
+    const firstProvider = [...this._activeProviders][0];
+    const provider = this._providers.get(firstProvider);
+
+    let renderer =
+      provider?.renderer ??
+      Completer.getDefaultRenderer(newCompleterContext.sanitizer);
+    const modelFactory = provider?.modelFactory;
+    let model: Completer.IModel;
+    if (modelFactory) {
+      model = await modelFactory.call(provider, newCompleterContext);
+    } else {
+      model = new CompleterModel();
+    }
+
+    this._mostRecentContext.set(widget.id, newCompleterContext);
     if (!handler) {
       // Create a new handler.
-      const handler = await this.generateHandler(newCompleterContext);
+      const handler = await this.generateHandler(
+        newCompleterContext,
+        model,
+        renderer
+      );
       this._panelHandlers.set(widget.id, handler);
       widget.disposed.connect(old => {
         this.disposeHandler(old.id, handler);
```

Comment:
The `old.id` also needs to be removed from the `_mostRecentContext` map.

Reply:
Thank you, this was addressed in https://github.com/jupyterlab/jupyterlab/pull/15065/commits/ecde7776852bc77764d1239ea17e82e6d4797983.